scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/templates/tests/barbican-functional.sh.j2

```diff
@@ -0,0 +1,110 @@
+export REQUESTS_CA_BUNDLE={{ ca_bundle }}
+
+set +u
+source ${VENV_FOLDER}/bin/activate
+set -u
+cd {{ ardana_qe_tests_dir }}
+
+${VENV_FOLDER}/bin/pip install --upgrade pip
+${VENV_FOLDER}/bin/pip install -r requirements.txt
+${VENV_FOLDER}/bin/pip install -r test-requirements.txt
+
+if [ ! -d {{ ardana_qe_tests_dir }}/.testrepository ]; then
+    ${VENV_FOLDER}/bin/testr init
+fi
+
+if [ ! -d /var/lib/ardana/.barbican ]; then
+    mkdir /var/lib/ardana/.barbican
+fi
+
+cat >/var/lib/ardana/.barbican/barbican.conf <<EOF
+[DEFAULT]
+
+EOF
+
+cat >{{ ardana_qe_tests_dir }}/etc/barbican/barbican-functional.conf <<EOF
+[DEFAULT]
+[identity]
+uri={{ os_admin_osrc.OS_AUTH_URL }}
+version=v3
+username={{ username }}
+project_name={{ project_name }}
+password={{ password }}
+domain_name={{ os_admin_osrc.OS_DOMAIN_NAME }}
+service_admin={{ service_admin }}
+service_admin_project={{ service_admin_project }}
+service_admin_password={{ service_admin_password }}
+
+[keymanager]
+service_type={{ service_type }}
+service_name={{ service_name }}
+region_name={{ region_name }}
+endpoint_type={{ endpoint_type }}
+server_host_href_set={{ server_host_href_set }}
+
+[rbac_users]
+project_a={{ project_a }}
+project_b={{ project_b }}
+admin_a={{ admin_a }}
+admin_a_password={{ admin_a_password }}
+creator_a={{ creator_a }}
+creator_a_password={{ creator_a_password }}
+creator_a_2={{ creator_a_2 }}
+creator_a_2_password={{ creator_a_2_password }}
+observer_a={{ observer_a }}
+observer_a_password={{ observer_a_password }}
+auditor_a={{ auditor_a }}
+auditor_a_password={{ auditor_a_password }}
+admin_b={{ admin_b }}
+admin_b_password={{ admin_b_password }}
+creator_b={{ creator_b }}
+creator_b_password={{ creator_b_password }}
+observer_b={{ observer_b }}
+observer_b_password={{ observer_b_password }}
+auditor_b={{ auditor_b }}
+auditor_b_password={{ auditor_b_password }}
+
+[quotas]
+quota_secrets={{ quota_secrets }}
+quota_orders={{ quota_orders }}
+quota_containers={{ quota_containers }}
+quota_consumers={{ quota_consumers }}
+quota_cas={{ quota_cas }}
+
+EOF
+
+cat >{{ ardana_qe_tests_dir }}/.testr.conf <<EOF
+[DEFAULT]
+test_command=OS_STDOUT_CAPTURE=${OS_STDOUT_CAPTURE:-1} \
+             OS_STDERR_CAPTURE=${OS_STDERR_CAPTURE:-1} \
+             OS_TEST_TIMEOUT=${OS_TEST_TIMEOUT:-500} \
+             OS_TEST_LOCK_PATH=${OS_TEST_LOCK_PATH:-${TMPDIR:-'/tmp'}} \
+             ${PYTHON:-python} -m subunit.run discover -t ${OS_TOP_LEVEL:-./} ${OS_TEST_PATH:-./functionaltests/api} \$LISTOPT \$IDOPTION
+test_id_option=--load-list \$IDFILE
+test_list_option=--list
+group_regex=([^\.]*\.)*
+
+EOF
+
+cp /var/lib/ardana/openstack/ardana/ansible/roles/tempest/files/tests2skip.py {{ ardana_qe_tests_dir }}/tests2skip.py
+
+http_proxy= https_proxy= HTTP_PROXY= HTTPS_PROXY= ${TESTR} run \
+    --subunit $($PYTHON ${TESTS_TO_SKIP_BIN} ${RUN_FILTER})|
+    ${VENV_FOLDER}/bin/subunit-trace --no-failure-debug -f | tee testr_results.log
```

Comment:
Please save the test output to `{{ ardana_qe_test_log }}` so jenkins can store the test log.